modules/effects/arrow-effects-rx2/src/main/kotlin/arrow/effects/MaybeK.kt

```diff
@@ -0,0 +1,83 @@
+package arrow.effects
+
+import arrow.core.*
+import arrow.effects.typeclasses.Proc
+import arrow.higherkind
+import io.reactivex.Maybe
+import io.reactivex.MaybeEmitter
+
+fun <A> Maybe<A>.k(): MaybeK<A> = MaybeK(this)
+
+fun <A> MaybeKOf<A>.value(): Maybe<A> = this.fix().maybe
+
+@higherkind
+data class MaybeK<A>(val maybe: Maybe<A>) : MaybeKOf<A>, MaybeKKindedJ<A> {
+
+  fun <B> map(f: (A) -> B): MaybeK<B> =
+    maybe.map(f).k()
+
+  fun <B> ap(fa: MaybeKOf<(A) -> B>): MaybeK<B> =
+    flatMap { a -> fa.fix().map { ff -> ff(a) } }
+
+  fun <B> flatMap(f: (A) -> MaybeKOf<B>): MaybeK<B> =
+    maybe.flatMap { f(it).fix().maybe }.k()
+
+  fun <B> fold(ifEmpty: () -> B, ifSome: (A) -> B): B = when {
```

Comment:
According to RxJava's docs, `this.maybe.blockingGet()` returns `null` for no value and `T` for a value. Given that RxJava2 doesn't allow nulls in `Maybe`, you can rewrite this to be simpler:

maybe.blockingGet().let {
  if (it == null) ifEmpty() else ifSome(if)
}

Reply:
I would suggest adding a new test for it too.